build/lib/src/resource/resource.dart

```diff
@@ -53,8 +60,16 @@ class Resource<T> {
 class ResourceManager {
   final _resources = new Set<Resource>();
 
+  /// The [Resource]s that we need to call `beforeExit` on. We have to hang on
+  /// to these forever, but they should be small in number, and we don't hold
+  /// on to the instances.
```

Comment:
"we don't hold on to the instances" - this isn't clear to me, could you rephrase to make it more clear we mean the instances of the created resource, not the `Resource` instance?

Reply:
done